### PR TITLE
tests: ensure snap-confine apparmor profile is parsable

### PR DESCRIPTION
**Pull request by mvo5**
We had the issue that we did a 'include "..."' in our snap-confine
apparmor profile. However the apparmor tools do not support
include, they do support "#include" though. To catch issues like
this we now check explicitely in the snap-confine spread test.

### CODE REVIEW

File: tests/main/snap-confine/task.yaml [hunk 1]
```diff
@@ -39,3 +39,15 @@ execute: |
     echo "But nvidia files are gone"
     ! test -f /run/udev/tags/snap_test-snapd-tools_echo/+module:nvidia
     ! test -f /run/udev/tags/snap_test-snapd-tools_echo/+module:nvidia_modeset
+
+    echo "Ensure apparmor profile for snap-confine is parsable"
+    if which aa-enforce 2>/dev/null; then
+        aa-enforce
```
Comment (zyga) [inline tests/main/snap-confine/task.yaml]:
```
Ensure apparmor profile for snap-confine is parsable
+ which aa-enforce
/usr/sbin/aa-enforce
+ aa-enforce
usage: aa-enforce [-h] [-d DIR] [--no-reload] program [program ...]
aa-enforce: error: the following arguments are required: program
-----
```

File: tests/main/snap-confine/task.yaml [hunk 2]
```diff
@@ -39,3 +39,12 @@ execute: |
     echo "But nvidia files are gone"
     ! test -f /run/udev/tags/snap_test-snapd-tools_echo/+module:nvidia
     ! test -f /run/udev/tags/snap_test-snapd-tools_echo/+module:nvidia_modeset
+
+    echo "Ensure apparmor profile for snap-confine is parsable"
+    for f in /etc/apparmor.d/usr.lib.snapd.snap-confine*; do
+        apparmor_parser -QTK $f
+        if which aa-enforce 2>/dev/null; then
+            aa-enforce
```
Comment (bboozzoo) [inline tests/main/snap-confine/task.yaml]: This one should probably be removed as well.

File: tests/main/snap-confine/task.yaml [hunk 3]
```diff
@@ -39,3 +39,12 @@ execute: |
     echo "But nvidia files are gone"
     ! test -f /run/udev/tags/snap_test-snapd-tools_echo/+module:nvidia
     ! test -f /run/udev/tags/snap_test-snapd-tools_echo/+module:nvidia_modeset
+
+    echo "Ensure apparmor profile for snap-confine is parsable"
```
Comment (bboozzoo) [inline tests/main/snap-confine/task.yaml]: You'll need to skip this part for fedora as there's no AppArmor. Looking at the tests, it's likely that debian-9 image is not using a kernel that supports AppArmor

Comment (zyga) [reply tests/main/snap-confine/task.yaml]: You can always parse the profile and just not load it into the kernel. `apparmor_parser --skip-kernel-load`

Comment (sergiocazzolato) [reply tests/main/snap-confine/task.yaml]: On debian also failing. 
ERROR: Cache read/write disabled: interface file missing. (Kernel needs AppArmor 2.4 compatibility patch.)

Comment (sergiocazzolato) [reply tests/main/snap-confine/task.yaml]: In fedora it is failing because the command apparmor_parser does not exist

**Comment by bboozzoo** [comment]
The tests don't start:
```
TZ=UTC
USER=travis
travis_fold:end:env
spread
2017-12-11 10:50:23 Found /home/travis/gopath/src/github.com/snapcore/snapd/spread.yaml.
error: tests/main/snap-confine specifies systems both in delta and plain format
```

File: tests/main/snap-confine/task.yaml [hunk 4]
```diff
@@ -1,7 +1,8 @@
 summary: Test that snap-confine errors in the right way
 
 # the error message can only happen on classic systems
-systems: [-ubuntu-core-16-*]
+# our debian image does not have fully working apprmor
+systems: [-ubuntu-core-16-*, debian-*]
```
Comment (bboozzoo) [inline tests/main/snap-confine/task.yaml]: `-debian-*` ?

**Comment by bboozzoo** [comment]
merge?

**Pull request merged**